GlimpseImputationPipeline/Glimpse2MergeBatches.wdl

```diff
@@ -23,33 +26,88 @@ workflow Glimpse2MergeBatches {
+                call SubsetToIntervals {
+                    input:
+                        vcf = imputed_vcfs[batch_index],
+                        vcf_index = imputed_vcf_indices[batch_index],
+                        interval_list = intervals,
+                        docker_gatk = docker_gatk
+                }
+                call ExtractAnnotations {
+                    input:
+                        imputed_vcf = SubsetToIntervals.subset_vcf,
+                        imputed_vcf_index = SubsetToIntervals.subset_vcf_index,
+                        batch_index = batch_index,
+                        docker_extract_annotations = docker_gatk
+                }    
+            }
+
+            call MergeVcfs {
                 input:
-                    imputed_vcf = imputed_vcfs[batch_index],
-                    imputed_vcf_index = imputed_vcf_indices[batch_index],
-                    batch_index = batch_index,
-                    docker_extract_annotations = docker_extract_annotations
+                    imputed_vcfs = SubsetToIntervals.subset_vcf,
+                    imputed_vcf_indices = SubsetToIntervals.subset_vcf_index,
+                    docker_merge = docker_merge,
+                    mem_gb = mem_gb_merge
             }
-            call CountSamples {
+
+            call RecomputeAndAnnotate {
                 input:
-                    imputed_vcf = imputed_vcfs[batch_index],
-                    imputed_vcf_index = imputed_vcf_indices[batch_index],
-                    docker_count_samples = docker_count_samples
+                    merged_vcf = MergeVcfs.merged_vcf,
+                    annotations = ExtractAnnotations.annotations,
+                    num_samples = CountSamples.num_samples,
+                    output_basename = "merged_annotated",
+                    mem_gb = mem_gb_merge,
+                    docker_merge = docker_merge
             }
         }
 
-        call MergeAndRecomputeAndAnnotate {
+        call GatherVcfs {
+            input:
+                input_vcfs = RecomputeAndAnnotate.merged_imputed_vcf,
+                output_vcf_name = output_basename + ".vcf.gz"
+        }
+
+        call CountSamples as CountSamplesFinal {
             input:
-                imputed_vcfs = imputed_vcfs,
-                imputed_vcf_indices = imputed_vcf_indices,
-                qc_metrics = qc_metrics,
-                annotations = ExtractAnnotations.annotations,
-                num_samples = CountSamples.num_samples,
-                output_basename = output_basename,
-                mem_gb = mem_gb_merge,
-                docker_merge = docker_merge
+                imputed_vcf = GatherVcfs.output_vcf,
+                imputed_vcf_index = GatherVcfs.output_vcf_index,
+                docker_count_samples = docker_count_samples
+        }
+
+        if (defined(qc_metrics)) {
+            call MergeQCMetrics {
+                input:
+                    qc_metrics = select_first([qc_metrics]),
```

Comment:
This `select_first` will fail if qc_metrics is null. I see two options:
1. You can make QC metrics a required input to this WDL. It initially wasn't because we said that QC collection for imputation should be optional, but at this point we might as well always turn it on.
2. Make qc_metrics an optional input to the `MergeQCMetrics` task, and then handle the null case there differently (see comment for line 472)

Reply:
isn't this fine, because we'll only get into this block if qc_metrics is not null?

that said, I think your point about just making qc collection required makes sense, so going to take that route.